src/lib/server/hyrise_communicator.cpp

```diff
@@ -0,0 +1,111 @@
+  std::unique_ptr<SQLPipeline> sql_pipeline;
+  try {
+    sql_pipeline = std::make_unique<SQLPipeline>(SQLPipelineBuilder{sql}.create_pipeline());
+    sql_pipeline->get_result_table();
+  } catch (const InvalidInputException& exception) {
+    execution_info.error = exception.what();
+    return execution_info;
+  }
+
+  const auto [pipeline_status, result_table] = sql_pipeline->get_result_table();
+  if (pipeline_status == SQLPipelineStatus::Success) {
+    execution_info.result_table = result_table;
+    execution_info.root_operator = sql_pipeline->get_physical_plans().front()->type();
+
+    if (debug_note) {
+      std::stringstream stream;
+      stream << sql_pipeline->metrics();
+      execution_info.execution_information = stream.str();
+    }
+  } else {
+    const std::string failed_statement = sql_pipeline->failed_pipeline_statement()->get_sql_string();
+    execution_info.error = "Error during pipeline execution! Failed statement: " + failed_statement;
+  }
+  return execution_info;
+}
+
+std::optional<std::string> HyriseCommunicator::setup_prepared_plan(const std::string& statement_name,
+                                                                   const std::string& query) {
+  // Named prepared statements must be explicitly closed before they can be redefined by another Parse message
+  // https://www.postgresql.org/docs/10/static/protocol-flow.html
+  if (Hyrise::get().storage_manager.has_prepared_plan(statement_name)) {
+    AssertInput(statement_name.empty(),
+                "Named prepared statements must be explicitly closed before they can be redefined.");
+    Hyrise::get().storage_manager.drop_prepared_plan(statement_name);
+  }
+
+  auto pipeline_statement = SQLPipelineBuilder{query}.create_pipeline_statement();
+  auto sql_translator = SQLTranslator{UseMvcc::Yes};
+  auto prepared_plans = std::vector<std::shared_ptr<AbstractLQPNode>>();
+  try {
+    prepared_plans = sql_translator.translate_parser_result(*pipeline_statement.get_parsed_sql_statement());
+  } catch (const InvalidInputException& exception) {
+    return std::string(exception.what());
+  }
+  Assert(prepared_plans.size() == 1u, "Only a single statement allowed in prepared statement");
+
+  const auto prepared_plan =
+      std::make_shared<PreparedPlan>(prepared_plans[0], sql_translator.parameter_ids_of_value_placeholders());
+
+  Hyrise::get().storage_manager.add_prepared_plan(statement_name, std::move(prepared_plan));
+  return {};
+}
+
+std::pair<std::string, std::shared_ptr<AbstractOperator>> HyriseCommunicator::bind_prepared_plan(
+    const PreparedStatementDetails& statement_details) {
+  Assert(Hyrise::get().storage_manager.has_prepared_plan(statement_details.statement_name),
+         "The specified statement does not exist.");
+
+  const auto prepared_plan = Hyrise::get().storage_manager.get_prepared_plan(statement_details.statement_name);
+
+  if (statement_details.statement_name.empty()) {
```

Comment:
What are you doing here with that conditional?

Reply:
This is already done at parsing step. Removing this piece of code.